src/DRAMPower/DRAMPower/standards/lpddr5/LPDDR5.cpp

```diff
@@ -47,58 +47,121 @@ namespace DRAMPower {
 
     void LPDDR5::registerPatterns() {
         using namespace pattern_descriptor;
-        // All commands consider 16B bank architecture mode
-        // ---------------------------------:
-
         // LPDDR5 needs 2 commands for activation (ACT-1 and ACT-2)
         // ACT-1 must be followed by ACT-2 in almost every case (CAS, WRITE,
         // MASK WRITE and READ commands can be issued inbetween ACT-1 and ACT-2)
         // Here we consider ACT = ACT-1 + ACT-2, not considering interleaving
-        this->registerPattern<CmdType::ACT>({
+        commandPattern_t act_pattern = {
             // ACT-1
             H, H, H, R14, R15, R16, R17,
             BA0, BA1, BA2, BA3, R11, R12, R13,
             // ACT-2
             H, H, L, R7, R8, R9, R10,
             R0, R1, R2, R3, R4, R5, R6
-        });
-        this->registerPattern<CmdType::PRE>({
+        };
+        if (memSpec.bank_arch == MemSpecLPDDR5::BG) {
+            act_pattern[9] = BG0;
+            act_pattern[10] = BG1;
+        } else if (memSpec.bank_arch == MemSpecLPDDR5::B8) {
+            act_pattern[10] = V;
+        }
+        this->registerPattern<CmdType::ACT>(act_pattern);
+
+        commandPattern_t pre_pattern = {
             L, L, L, H, H, H, H,
             BA0, BA1, BA2, BA3, V, V, L
-        });
-        this->registerPattern<CmdType::PREA>({
+        };
+        if (memSpec.bank_arch == MemSpecLPDDR5::BG) {
+            pre_pattern[9] = BG0;
+            pre_pattern[10] = BG1;
+        } else if (memSpec.bank_arch == MemSpecLPDDR5::B8) {
+            pre_pattern[10] = V;
+        }
+        this->registerPattern<CmdType::PRE>(pre_pattern);
+
+        commandPattern_t prea_pattern = {
```

Comment:
For PREA and REFA the bank address is a don't care, so we could also assign V.

Reply:
done